tests/cases/fourslash/findAllRefsRootSymbols.ts

```diff
@@ -0,0 +1,17 @@
+/// <reference path="fourslash.ts" />
+
+////interface I { [|{| "isWriteAccess": true, "isDefinition": true |}x|]: {}; }
+////interface J { [|{| "isWriteAccess": true, "isDefinition": true |}x|]: {}; }
+////declare const o: (I | J) & { [|{| "isWriteAccess": true, "isDefinition": true |}x|]: string };
+////o.[|x|];
+
+const [r0, r1, r2, r3] = test.ranges();
+verify.referenceGroups(r0, [{ definition: "(property) I.x: {}", ranges: [r0, r3] }]);
+verify.referenceGroups(r1, [{ definition: "(property) J.x: {}", ranges: [r1, r3] }]);
+verify.referenceGroups(r2, [{ definition: "(property) x: string", ranges: [r2, r3] }]);
+verify.referenceGroups(r3, [
+	{ definition: "(property) I.x: {}", ranges: [r0] },
+	{ definition: "(property) J.x: {}", ranges: [r1] },
+	{ definition: "(property) x: string", ranges: [r2] },
+	{ definition: "(property) x: string & {}", ranges: [r3] },
```

Comment:
how did this malfunction before this fix? were the first two references missing?

Reply:
Currently, every `x` has no references except to itself, because we can't figure that they're all the same thing. `getRootSymbols` returns some transient symbols that aren't helpful.
You can test it out by pasting this in a file:
```ts
interface I { x: {}; }
interface J { x: {}; }
declare const o: (I | J) & { x: string };
o.x;
```

Reply:
@sandersn Bump -- good to merge?